### ISSUE TITLE
Portability issue in configure script

### ISSUE THREAD

**Issue by tbvdm** [sqlcipher/sqlcipher#382]
I ran the configure script (from tag v4.4.3) on OpenBSD and noticed the following in its output:

```
checking for crypto library to use... ./configure[12165]: CFLAGS+= -DSQLCIPHER_CRYPTO_OPENSSL: not found
./configure[12165]: BUILD_CFLAGS+= -DSQLCIPHER_CRYPTO_OPENSSL: not found
openssl
```

This is because the configure script uses += to append to a string variable, which is a bashism and not portable. The fix would be straightforward: in configure.ac, replace every instance of `FOO+=" bar"` with `FOO="$FOO bar"`.

(I'd submit a PR but I understand I'd need to sign an agreement first.)

**Comment by developernotes** [comment]
Hello @tbvdm 

Thanks for your interest in SQLCipher and for identifying a build issue for OpenBSD. We will make the adjustment on our side and include it with the next release of SQLCipher. Thanks again!

**Comment by tbvdm** [comment]
Great, thanks. :)

**Comment by sjlombardo** [comment]
This issue has been resolved as of the SQLCipher release 4.5.0
